guides/typescript.md

```diff
@@ -0,0 +1,41 @@
+# TypeScript
+
+These guidelines specifically apply to TypeScript.
+
+## Provide explicit return types for functions/methods
```

Comment:
Good point! I've added the [`explicit-function-return-type`](https://typescript-eslint.io/rules/explicit-function-return-type/) rule here: https://github.com/MetaMask/eslint-config/pull/314

Reply:
By the way, given that we have some ESLint rules that we've enabled ourselves, what do we feel about explicitly documenting the reason for doing so? That means we'd keep this here but then have something at the top that says something like:

``` markdown
> **Note**  
> This guideline is enforced via [`explicit-function-return-type`](/link/to/rule/in/our/eslint-config).
```

Reply:
That seems like a good idea to me. We can remove it later if we don't think it has been useful.

Generally it would be nice to keep this guide shorter, and let ESLint explain things for us where possible. But the ESLint rules don't explain our reasoning very well. I wish we could.... explain the reasons for lint rules inline and advice a course of action, something like that. Ah well.

Reply:
Agreed... okay. I can add this in a followup PR once we have a new version of `eslint-config` out.